sql/index/pilosa/driver.go

```diff
@@ -67,19 +67,22 @@ type (
 		timeMapping time.Duration
 	}
 
+	tableMap map[string]*pilosa.Holder
```

Comment:
I believe we don't need to keep a holders' lookup table here. We can create/open the holder on demand in every function where is needed.

Reply:
Are you sure? I believe it's better to have them loaded when the server starts. Loading the indexes each time they are used seem to be very inefficient. Maybe I'm misunderstanding something.

Reply:
Generally we don't use the `Holder` in iterators. Only at the beginning when you `Load` and when you Create/Save. It's like we had `SyncSchema` in old pilosa driver. 
I believe we'll not notice any performance regression. And if we do, then we can think about optimization.

Reply:
It's almost the same thing what we have with mapping.

Reply:
I'll try that way, thanks.

Reply:
I've doing this in two ways:

* Create new holder in create and load but maintain holder in driver struct: 
https://github.com/jfontan/go-mysql-server/commit/fbd22ab940152cb4cfc1052be2c5f38962a3ff9d
* Create new holders when needed and passing it to child (`loadIndex`): https://github.com/jfontan/go-mysql-server/commit/2f4c75f189fb176666ab4f57850200989b5130b7

In both cases this leads to index fields to contain an empty map, resulting in a panic:

```
 go test -v gopkg.in/src-d/go-mysql-server.v0/sql/index/pilosa -run TestIntersection -count 1
=== RUN   TestIntersection
--- FAIL: TestIntersection (0.00s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x94aac7]

goroutine 6 [running]:
testing.tRunner.func1(0xc0001aa100)
	/usr/lib/go/src/testing/testing.go:792 +0x387
panic(0x9fcfe0, 0xfb0bd0)
	/usr/lib/go/src/runtime/panic.go:513 +0x1b9
github.com/pilosa/pilosa.(*Field).Name(...)
	/home/jfontan/go/src/github.com/pilosa/pilosa/field.go:188
gopkg.in/src-d/go-mysql-server.v0/sql/index/pilosa.(*indexLookup).values(0xc0002ce8c0, 0xb45de0, 0xb40608, 0x0, 0x0, 0x0)
	/home/jfontan/go/src/gopkg.in/src-d/go-mysql-server.v0/sql/index/pilosa/lookup.go:93 +0x177
gopkg.in/src-d/go-mysql-server.v0/sql/index/pilosa.(*indexLookup).values(0xc0002ce930, 0xb45de0, 0xb40608, 0x0, 0x0, 0x0)
	/home/jfontan/go/src/gopkg.in/src-d/go-mysql-server.v0/sql/index/pilosa/lookup.go:121 +0x300
gopkg.in/src-d/go-mysql-server.v0/sql/index/pilosa.(*indexLookup).Values(0xc0002ce930, 0xb45de0, 0xb40608, 0x0, 0x0, 0x0, 0x0)
	/home/jfontan/go/src/gopkg.in/src-d/go-mysql-server.v0/sql/index/pilosa/lookup.go:137 +0x9c
gopkg.in/src-d/go-mysql-server.v0/sql/index/pilosa.TestIntersection(0xc0001aa100)
	/home/jfontan/go/src/gopkg.in/src-d/go-mysql-server.v0/sql/index/pilosa/driver_test.go:410 +0xc0d
testing.tRunner(0xc0001aa100, 0xacef50)
	/usr/lib/go/src/testing/testing.go:827 +0xbf
created by testing.(*T).Run
	/usr/lib/go/src/testing/testing.go:878 +0x353
FAIL	gopkg.in/src-d/go-mysql-server.v0/sql/index/pilosa	0.008s
```

I still don't understand why it works when the holders are kept in a map but not when they are not.